src/python/pants/backend/adhoc/target_types.py

```diff
@@ -297,6 +299,58 @@ def enum_value(self) -> PathEnvModifyMode:
         return PathEnvModifyMode(self.value)
 
 
+class AdhocToolOutputsMatchErrorBehavior(StringField):
+    alias = "outputs_match_error_behavior"
+    default = GlobMatchErrorBehavior.error.value
+    help = help_text(
+        """
+        Configure whether to check the outputs expected to be captured by this `adhoc_tool` were
+        in fact actually captured.
+
+        Valid values are:
+
+        - `error`: Raise an error if none of the proided globs matched.
+        - `warn`: Log a warning if none of the provided globs matched.
```

Comment:
Should there not also be an `ignore`? It may be perfectly valid for a file to be optional.

Reply:
Oh, I guess `allow_empty` is that. Never mind!